woocommerce/woocommerce-android
# Unified Login: wpcom flow

Closes #2779 by implementing the logic for login with wordpress.com. This PR includes the following changes:
- Login with wpcom email + password
- Login with wpcom email + magic link
- Login with wpcom -> Google signin
- Site Picker styling revamped

Not included:
- Error screens will be handled in a separate PR.
- The store picker screen is completely styled per the designs as I have some open questions about that design and will handle the feedback in a separate PR. 

### Screenshots

Prologue screen | wpcom email screen | Store picker screen
-- | -- | --
![Screenshot_1598653334](https://user-images.githubusercontent.com/5810477/91623310-f553c580-e968-11ea-8b26-4aedd0bab217.png)|![Screenshot_1598653338](https://user-images.githubusercontent.com/5810477/91623313-f84eb600-e968-11ea-9883-7c9da4032989.png)|![Screenshot_1598653462](https://user-images.githubusercontent.com/5810477/91623315-fab11000-e968-11ea-9200-aee7e6f3dd63.png)

### To Test
- Test logging in with a wordpress.com account using the password
- Test logging in with a wordpress.com account using magic link
- Test logging in with a wordpress.com account then selecting to login with google
- Test logging in with site address still works as expected


Update release notes:
